test/tmetrics.m

```diff
@@ -0,0 +1,349 @@
+            mt = p.getMeter(metername);
+            ct = mt.createCounter(countername);
+
+            % verify MATLAB object properties
+            verifyEqual(testCase, mt.Name, metername);
+            verifyEqual(testCase, mt.Version, "");
+            verifyEqual(testCase, mt.Schema, "");
+            verifyEqual(testCase, ct.Name, countername);
+
+            % create testing value 
+            val = 10;
+
+            % add value and attributes
+            ct.add(val);
+
+            % wait for default collector response time (60s)
+            pause(70);
+
+            % fetch result
+            results = readJsonResults(testCase);
+            results = results{1};
+
+            % verify meter and counter names
+            verifyEqual(testCase, string(results.resourceMetrics.scopeMetrics.metrics.name), countername);
+            verifyEqual(testCase, string(results.resourceMetrics.scopeMetrics.scope.name), metername);
+
+            % fetch datapoint
+            dp = results.resourceMetrics.scopeMetrics.metrics.sum.dataPoints;
+
+            % verify counter value
+            verifyEqual(testCase, dp.asDouble, val);
+
+        end
+
+
+        function testCounterAddAttributes(testCase)
+            % test names, added value and attributes in Counter
+
+            metername = "foo";
+            countername = "bar";
+
+            p = opentelemetry.sdk.metrics.MeterProvider();
+            mt = p.getMeter(metername);
+            ct = mt.createCounter(countername);
+
+            % verify MATLAB object properties
+            verifyEqual(testCase, mt.Name, metername);
+            verifyEqual(testCase, mt.Version, "");
+            verifyEqual(testCase, mt.Schema, "");
+            verifyEqual(testCase, ct.Name, countername);
+
+            % create testing value and dictionary
+            dict = dictionary("k1","v1","k2",5);
+            vals = [1,2,3];
+            dict_keys = keys(dict);
+            dict_vals = values(dict);
+
+            % add value and attributes
+            ct.add(vals(1),dict);
+            ct.add(vals(2),dict);
+            ct.add(vals(3),dict_keys(1),dict_vals(1),dict_keys(2),dict_vals(2));
+
+            % wait for default collector response time (60s)
+            pause(70);
+
+            % fetch result
+            results = readJsonResults(testCase);
+            results = results{1};
+
+            % verify meter and counter names
+            verifyEqual(testCase, string(results.resourceMetrics.scopeMetrics.metrics.name), countername);
+            verifyEqual(testCase, string(results.resourceMetrics.scopeMetrics.scope.name), metername);
+
+            % fetch datapoint
+            dp = results.resourceMetrics.scopeMetrics.metrics.sum.dataPoints;
+
+            % verify counter value
+            verifyEqual(testCase, dp.asDouble, sum(vals));
+
+            % verify counter attributes
+            verifyEqual(testCase, string(dp.attributes(1).key), dict_keys(1));
```

Comment:
These seems fragile in the sense that it relies on the results keeping the attributes in the same order. Refer to testAttributes in ttrace.m for a way to write tests that does not assume the attribute order.